bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHABaseHandler.java

```diff
@@ -0,0 +1,201 @@
+                return;
+            }
+
+            if (bridge.getStatus() == OFFLINE) {
+                updateStatus(OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                        "The Plugwise Home Automation bridge is currently offline.");
+            }
+
+            initialize(config);
+        } else {
+            logger.warn("Invalid config for Plugwise Home Automation thing handler with config = {}", config);
+        }
+    }
+
+    @Override
+    public final void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Handling command = {} for channel = {}", command, channelUID);
+
+        if (getThing().getStatus() == ONLINE) {
+            PlugwiseHAController controller = getController();
+            if (controller != null) {
+                try {
+                    E entity = getEntity(controller, false);
+                    if (entity != null) {
+                        if (this.isLinked(channelUID)) {
+                            if (command instanceof RefreshType) {
+                                refreshChannel(entity, channelUID);
+                            } else {
+                                handleCommand(entity, channelUID, command);
+                            }
+                        }
+                    }
+                } catch (PlugwiseHAException e) {
+                    logger.warn("Unexpected error handling command = {} for channel = {} : {}", command, channelUID,
+                            e.getMessage());
+                }
+            }
+        }
+    }
+
+    // Public member methods
+
+    public @Nullable PlugwiseHABridgeHandler getPlugwiseHABridge() {
+        Bridge bridge = this.getBridge();
+        if (bridge != null) {
+            return (PlugwiseHABridgeHandler) bridge.getHandler();
+        }
+
+        return null;
+    }
+
+    @SuppressWarnings("unchecked")
+    public C getPlugwiseThingConfig() {
+        return (C) getConfigAs(clazz);
+    }
+
+    // Private & protected methods
+
+    private final @Nullable PlugwiseHAController getController() {
+        Bridge bridge = getBridge();
+
+        if (bridge != null) {
+            if (bridge.getHandler() instanceof PlugwiseHABridgeHandler) {
+                PlugwiseHABridgeHandler handler = (PlugwiseHABridgeHandler) bridge.getHandler();
+                return handler.getController();
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Checks the configuration for validity, result is reflected in the status of
+     * the Thing
+     */
+    private boolean checkConfig(C config) {
+        if (config == null || !config.isValid()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Configuration is missing or corrupted");
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    protected final void refresh() {
+        if (getThing().getStatus() == ONLINE) {
```

Comment:
You might want to override `bridgeStatusChanged` in this class to set the status in these things if the bridge goes offline, and set the status to ONLINE if OFFLINE. See some other bindings in relation to overriding `bridgeStatusChanged`.

Reply:
Not sure what you mean. Could you give some more hints?

Reply:
Yes this is a big vague :smile: . What I meant, if I read it correctly, is that if the bridge is offline the bridge will set the childhandlers offline. However when offline you still might want to set the channels to something like undef. Because otherwise the channels will keep showing the last value, and than the user might not see somethings is not right by just looking at the values.  Also it can be helpful to set a thing offline when there is a except. But than if there is new data and it's all ok again the thing has to be set online again.

Reply:
If bindings should set all channels to `UnDefType.UNDEF` when the bridge goes offline, shouldn't that be the default behavior of `bridgeStatusChanged`?